File: Benchmarking/stats/euler8/euler8.py
import sys

path = sys.argv[1]

def adja(str):
	i = 0
	currentProd = 1
	maxProd = 0
	while i < len(str)-13:
		for j in range(0, 13):
			currentProd = currentProd * int(str[i+j])
		if currentProd > maxProd:
			maxProd = currentProd
		currentProd = 1
		i += 1
	return maxProd

def calc(path):
	maxi = 0
	with open(path, "r") as f:
		for l in f:
			temp = adja(l)
			if temp > maxi:
				maxi = temp
	f.close()
	return maxi

print calc(path)

File: Benchmarking/stats/euler8/euler8.java
import java.io.*;

public class euler8
{
	public static void main(String[] args)
	{
		String str;
		try
		(
			InputStream fis = new FileInputStream(args[0]);
			InputStreamReader isr = new InputStreamReader(fis);
			BufferedReader br = new BufferedReader(isr);
		)
		{
			String line = null;
			long maxi = 0;
			while((line = br.readLine()) != null)
			{
				long temp = adja(line);
				if(temp > maxi){
					maxi = temp;
				}
			}
			//System.out.println(maxi);
		}
		catch(Exception e)
		{
			System.out.println(e);
		}
	}

	private static long adja(String str)
	{
		long currentProd = 1;
		long maxProd = 0;
		for(int i = 0; i < str.length()-12; i++)
		{
			for(int j = 0; j < 13; j++)
			{
				currentProd = currentProd * Character.getNumericValue(str.charAt(i+j));
			}
			if(currentProd > maxProd)
			{
				maxProd = currentProd;
			}
			currentProd = 1;
		}
		return maxProd;
	}
}
